[VideoSubscriberAccount] Updates for Xcode13 Beta 1

tj-devel709 opened the pull request:
There were a lot of items inside the MacCatalyst todo file.
After running introspection tests, checking the web docs for all of them, and the xcode headers for most of them, 
it does not look like they are valid to be marked for MacCatalyst.

mandel-macaque reviewed (changes requested):
If there is an API that is not present in the platforms, it does not go to the ignore file, but we use a No*Attr, in all those, you should have used [NoMacCatalyst] like here: https://github.com/xamarin/xamarin-macios/blob/main/src/uikit.cs#L1519

tj-devel709 commented:
> If there is an API that is not present in the platforms, it does not go to the ignore file, but we use a No*Attr, in all those, you should have used [NoMacCatalyst] like here: https://github.com/xamarin/xamarin-macios/blob/main/src/uikit.cs#L1519

Ah so we add the NoMacCatalyst and remove it from the ignore?

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4885228&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/f514e85a7abb2a7e219942139b3ebc5b)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/a60ffb51e72d867560ae452428101f93)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/1db116f9ea7e6025652a37787854c55b)
* macCatiOS  [gist](https://gist.github.com/e79074b9bee5d74eb228c4c1facaa8eb)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/a84828ae7ff0f22d4ab211bf440e1d14)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/2fc49eb90b5d21f4137bafee3499f35d)
</details>

# API & Generator diff

:information_source: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/apigeneratordiff/;//api-diff.html) (please review changes)
:information_source: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210617.41/4885228/apigeneratordiff/;//generator-diff/index.html) (please review changes)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11977/4885228/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11977/4885228/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11977/4885228/apicomparison/api-diff.html)


# Test results

<details>
<summary>1 tests failed, 85 tests passed.</summary>

## Failed tests

 * introspection/watchOS 32-bits - simulator/Debug (watchOS 5.0): Failed
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4885228&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1094.BigSur'
Merge 8c0e292ab9c3ce419be1962d299f2963cfd36319 into abca8f1aec3d255147216a8e4aa95f7cbf13a554

spouliot commented:
Part of the issue is that 

```
	VideoSubscriberAccount \
```

is part of of `MACCATALYST_FRAMEWORKS =` inside `src/frameworks.sources`

So the bindings are included in Catalyst - and that means you must tell the generator not to process stuff that is not available on that platform.

Now looking back at the framework headers it seems that everything is not available at Catalyst, so you can remove its inclusion.

However to help "future you", the day Apple starts to include it, you better add the same `[NoMacCatalyst]` to new API when it's documented that way in the header files.

tj-devel709 commented:
@spouliot So after I removed VideoSubscriberAccount from the frameworks.sources and removed the ignore file, xtro still complains saying the items are not bound in MacCatalyst. So I still need the NoMacCatalyst right?

spouliot commented:
> xtro still complains saying the items are not bound in MacCatalyst

I'm not sure why xtro pick the API as missing :(
There might be a bad macro somewhere since the headers are clear

```
/// A VSAccountManager instance coordinates access to a subscriber's account.
API_AVAILABLE(ios(10.0), tvos(10.0)) API_UNAVAILABLE(macCatalyst)
@protocol VSAccountManagerDelegate <NSObject>
```

> So I still need the NoMacCatalyst right?

You should add it (like my previous comment about "future you") as it might become useful later. However that won't solve the problem.

Open an issue (for xtro) and add a link to it inside the `macCatalyst-VideoSubscriberAccount.ignore`. I'll have a look why later (but that might be after my vacations).

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4888190&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/6767d51d94e92542cd6c16ad6b40b565)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/b56db4681cb0a2785155d11a81fff0fc)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/51ed8901cedbe9e279df4f1c969e1757)
* macCatiOS  [gist](https://gist.github.com/3dba576b60eaa61a0ef0c69207c36d0a)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/16f263bc776b1d8deab81f687637b8a8)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/34b0d559db1a3f18fb3408290f836986)
</details>

# API & Generator diff

:information_source: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/apigeneratordiff/;//api-diff.html) (please review changes)
:information_source: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.17/4888190/apigeneratordiff/;//generator-diff/index.html) (please review changes)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11977/4888190/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11977/4888190/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11977/4888190/apicomparison/api-diff.html)


# Test results

<details>
<summary>3 tests failed, 83 tests passed.</summary>

## Failed tests

 * Xtro/Mac: Failed (Test run failed.)
 * link sdk/Mac Catalyst/Debug: TimedOut (Execution timed out after 1200 seconds.
No test log file was produced)
 * introspection/watchOS 32-bits - simulator/Debug (watchOS 5.0): Failed
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4888190&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1094.BigSur'
Merge bb6927351f3f7abea16c220bb6e59f6fdf1206e4 into aaf8d70573a4a242389501cbca66b2ca4c115c0e

tj-devel709 commented:
@spouliot @mandel-macaque 
I removed this API from the frameworks.sources under MacCatalyst
Then VSSubscriptionAccessLevel and VSSubscriptionAccess started complaining about not being in MacCatalyst so I added the NoMacCatalyst to those as well and added all the selectors xtro was complaining about to the ignore file as well as linked this issue here https://github.com/xamarin/maccore/issues/2462

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4890030&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/b3b92db2754e978c1017c4e0e60296d2)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/830a20223d870533e22afdc0522f07a2)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/e67d4950c3bb51afd9c936beefd71ca9)
* macCatiOS  [gist](https://gist.github.com/57993f520f0e989242cccbb5afd6c2d6)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/a7446946364055d7c8bc68d976538052)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/e8cc46e521c180f30081d62fd06d696e)
</details>

# API & Generator diff

:information_source: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/apigeneratordiff/;//api-diff.html) (please review changes)
:information_source: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210618.25/4890030/apigeneratordiff/;//generator-diff/index.html) (please review changes)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11977/4890030/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11977/4890030/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11977/4890030/apicomparison/api-diff.html)


# Test results

<details>
<summary>1 tests failed, 85 tests passed.</summary>

## Failed tests

 * introspection/watchOS 32-bits - simulator/Debug (watchOS 5.0): Failed
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4890030&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1104.BigSur'
Merge d43dc1c884e8b21bb0f43b7df040876f614fd186 into 4bc32a94ecb1c9af77c06d8e927ab6b3c4f2d1f5

rolfbjarne commented on src/videosubscriberaccount.cs:
```diff
@@ -103,10 +103,12 @@ interface IVSAccountManagerDelegate { }
 	interface VSAccountManagerDelegate {
 
 		[Abstract]
+		[NoMac]
```
This is a breaking change, it shows up as such in the `API & Generator diff`

Note that it seems you'll have to download the diff from Azure DevOps and hunt through the files to find it, because the links here don't work: https://xamarin.github.io/macios.ci/pr/PR11977/4890030/apicomparison/mac-api-diff.html

I think the best you can do is to add an `[Obsoleted (...)]` attribute, and hardcode an exception in introspection if there are any errors.

CC @mandel-macaque ^

tj-devel709 replied:
@rolfbjarne So I made these changes https://gist.github.com/tj-devel709/24bf5f179de2cb09184b6946f229637b based on other examples I found. Introspection does not seem to complain but Xtro is saying:
```
!extra-protocol-member! unexpected selector VSAccountManagerDelegate::accountManager:dismissViewController: found
!extra-protocol-member! unexpected selector VSAccountManagerDelegate::accountManager:presentViewController: found
```
Would I also add this to the ignore with a comment about marking it obsoleted?

mandel-macaque replied:
@rolfbjarne those links should work, is there a diff between API diff and Generator diff? I might have been cleaning too many files..

rolfbjarne replied:
@mandel-macaque open https://xamarin.github.io/macios.ci/pr/PR11977/4890030/apicomparison/mac-api-diff.html and try to click on any of the links there. Nothing happens.

rolfbjarne replied:
@tj-devel709 yes, this warrants an ignore addition to xtro.

tj-devel709 replied:
@rolfbjarne Okay, updated!

tj-devel709 replied:
I'm also not really sure how to read the Generator Diff.. Looks like the last iteration may have been complaining about me adding the NoMacCatalyst at certain places?

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4896355&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/31e134ac22add74f9ba8bfdb41059201)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/f3cd6c0e0f69c851a015a16a3e6a34c0)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/53be4ba6a1de95720dd9025b449e608a)
* macCatiOS  [gist](https://gist.github.com/fa76099f2f06afa6d4c69c6a95a6e9dd)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/ccb23c8b1f84ada5a82102db5efd198c)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/54c4582a4bf18001f60916d5f7644383)
</details>

# API & Generator diff

:information_source: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/apigeneratordiff/;//api-diff.html) (please review changes)
:information_source: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210621.20/4896355/apigeneratordiff/;//generator-diff/index.html) (please review changes)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11977/4896355/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11977/4896355/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11977/4896355/apicomparison/api-diff.html)


# Test results

<details>
<summary>2 tests failed, 84 tests passed.</summary>

## Failed tests

 * monotouch-test/Mac Catalyst/Debug [dotnet]: Failed (Tests run: 2622 Passed: 2489 Inconclusive: 35 Failed: 1 Ignored: 132)
 * Xtro/Mac: Failed (Test run failed.)
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4896355&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1100.BigSur'
Merge 4bef5c3efb29cd9f9c65d7d54b30e3454720fc7f into 6ce5f80cb7ef96000652fdd534c9176b793402e9

vs-mobiletools-engineering-service2 commented:
### :white_check_mark: [PR Build] Tests passed on Build. :white_check_mark:

Tests passed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4899351&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/cb6c3119e9499c264cbbf1f8781c4a6d)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/769e32514973afd5b36ff8a7ab9e2ebb)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/b225d4f69b4546ee4fa5e6d35d3447f6)
* macCatiOS  [gist](https://gist.github.com/153d58681a03d2ade558239a57eb859d)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/f6a2064d43e30f97accb398f199c0c7c)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/f9b4a582b8470467d535ed7f27ddecc3)
</details>

# API & Generator diff

:information_source: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/apigeneratordiff/;//api-diff.html) (please review changes)
:information_source: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210622.14/4899351/apigeneratordiff/;//generator-diff/index.html) (please review changes)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11977/4899351/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11977/4899351/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11977/4899351/apicomparison/api-diff.html)


# :tada: All 86 tests passed :tada:



[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4899351&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1101.BigSur'
Merge f3802ead991829f6596bdae9eb6b10a7e443259e into 0e07b989bd95c1760a7b8daac1a4050732c96e55
